trk/src/Screens/ClimbDetail/index.js

```diff
@@ -1,32 +1,74 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useContext } from 'react';
 import { Alert } from 'react-native';
 import TapsApi from '../../api/TapsApi';
+import ClimbsApi from '../../api/ClimbsApi';
 
 import { View, Text, StyleSheet, ScrollView, SafeAreaView, Image, TextInput, Button, TouchableWithoutFeedback, Keyboard, Share, TouchableOpacity } from 'react-native';
 
 import storage from '@react-native-firebase/storage';
 import SegmentedControl from '@react-native-segmented-control/segmented-control';
+import { AuthContext } from '../../Utils/AuthContext';
 
 
 function ClimbDetail(props) {
   console.log('[TEST] ClimbDetail called');
 
-  const { climbData } = props.route.params;
-  const { climbId } = props.route.params;
+  const { climbId, isFromHome } = props.route.params;
+  const [climbData, setClimbData] = useState(props.route.params.climbData || null);
+  const [isLoading, setIsLoading] = useState(isFromHome);
   const [climbImageUrl, setClimbImageUrl] = useState(null);
   const [setterImageUrl, setSetterImageUrl] = useState(null);
   const [completion, setCompletion] = useState('Zone');
   const [attempts, setAttempts] = useState('1');
   const [witness1, setWitness1] = useState('');
   const [witness2, setWitness2] = useState('');
+  const { currentUser } = useContext(AuthContext);
+  const [tapId, setTapId] = useState(props.route.params.tapId || null);
+
 
   const { navigation } = props;
 
+  useEffect(() => {
+    if (isFromHome && climbId) {
+      async function fetchData() {
+        try {
+          const climbDataResult = await ClimbsApi().getClimb(climbId);
+          if (climbDataResult && climbDataResult._data) {
+            setClimbData(climbDataResult._data);
+            console.log("Fetched climb data:", climbDataResult._data);
+  
+            if (currentUser.uid !== climbDataResult._data.setter) {
+              const { addTap } = TapsApi();
+              const tap = {
+                climb: climbId,
+                user: currentUser.uid,
+                timestamp: new Date(),
+                completion: 0,
+                attempts: '',
+                witness1: '',
+                witness2: '',
+              };
+              const documentReference = await addTap(tap);
+              setTapId(documentReference.id); // Set tapId only when navigating from home
+            } 
+          } else {
+            console.error('Climb data not found or is null');
+          }
+        } catch (error) {
+          console.error('Error fetching climb data:', error);
+        } finally {
+          setIsLoading(false);
+        }
+      }
+      fetchData();
+    }
+  }, [climbId, isFromHome, currentUser.uid]);
+  
+
   useEffect(() => {
```

Comment:
this used to be in home and used to be done before navigating to climbdetail. it
1. retrieves climbdata by using the id from the tag and climbsApi and calling firestore
2. creates a tap document if the user is not the setter of the climb